lib/ZwaveClient.ts

```diff
@@ -1022,41 +1044,44 @@ class ZwaveClient extends TypedEventEmitter<ZwaveClientEventCallbacks> {
 			}
 
 			// extend options with hidden `options`
-			const zwaveOptions: ZWaveOptions = Object.assign(
-				{
-					storage: {
-						cacheDir: storeDir,
-						deviceConfigPriorityDir: storeDir + '/config',
-					},
-					networkKey: this.cfg.networkKey,
-					logConfig: {
-						// https://zwave-js.github.io/node-zwave-js/#/api/driver?id=logconfig
-						enabled: this.cfg.logEnabled,
-						level: loglevels[this.cfg.logLevel],
-						logToFile: this.cfg.logToFile,
-						filename: ZWAVEJS_LOG_FILE,
-						forceConsole: true,
-						nodeFilter:
-							this.cfg.nodeFilter &&
-							this.cfg.nodeFilter.length > 0
-								? this.cfg.nodeFilter.map((n) => parseInt(n))
-								: undefined,
-					},
+			const zwaveOptions: utils.DeepPartial<ZWaveOptions> = {
+				storage: {
+					cacheDir: storeDir,
+					deviceConfigPriorityDir: priorityDir,
 				},
-				this.cfg.options
-			)
+				logConfig: {
+					// https://zwave-js.github.io/node-zwave-js/#/api/driver?id=logconfig
+					enabled: this.cfg.logEnabled,
+					level: loglevels[this.cfg.logLevel],
+					logToFile: this.cfg.logToFile,
+					filename: ZWAVEJS_LOG_FILE,
+					forceConsole: true,
+					nodeFilter:
+						this.cfg.nodeFilter && this.cfg.nodeFilter.length > 0
+							? this.cfg.nodeFilter.map((n) => parseInt(n))
+							: undefined,
+				},
+			}
+
+			if (this.cfg.scales) {
+				const scales: Record<string | number, string | number> = {}
```

Comment:
The reason this is modeled as an object is because you cannot define multiple preferred scales for one sensor type (or per named scale group like "temperature"). From a UX perspective, the multiselect dropdown seems wrong. Instead it should be: select sensor type or named scale, then select one preferred scale for that.